### ISSUE TITLE
kube.install fails

### ISSUE THREAD

**Issue by verizonold** [redhat-nfvpe/kube-ansible#201]
Hi 
I followed your instructions (http://dougbtv.com/nfvpe/2017/02/16/kubernetes-1.5-centos/) and created a master and 1 minion VM. Ran your kube install ansible script, but get the following errors. Can you please shed some light? I am using Virtual Box.

```
k: [kube-minion-1]
ok: [kube-master]

TASK [kube-install : Reload systemd units after changing 10-kubeadm.conf] *******************************
changed: [kube-minion-1]
changed: [kube-master]

TASK [kube-install : Start and enable kubelet] **********************************************************
changed: [kube-master]
changed: [kube-minion-1]

TASK [kube-install : Start docker when appropriate] *****************************************************
fatal: [kube-minion-1]: FAILED! => {"changed": false, "msg": "Unable to start service docker: Job for docker.service failed because the control process exited with error code. See \"systemctl status docker.service\" and \"journalctl -xe\" for details.\n"}
fatal: [kube-master]: FAILED! => {"changed": false, "msg": "Unable to start service docker: Job for docker.service failed because the control process exited with error code. See \"systemctl status docker.service\" and \"journalctl -xe\" for details.\n"}

PLAY RECAP **********************************************************************************************
kube-master                : ok=14   changed=2    unreachable=0    failed=1
kube-minion-1              : ok=14   changed=2    unreachable=0    failed=1
```

**Comment by verizonold** [comment]
I also see the following error in journalctl:

```
-- Unit kubelet.service has begun starting up.
Mar 20 02:12:08 localhost.localdomain kubelet[28754]: I0320 02:12:08.613484   28754 feature_gate.go:226] feature gates: &{{} map[]}
Mar 20 02:12:08 localhost.localdomain kubelet[28754]: I0320 02:12:08.613545   28754 controller.go:114] kubelet config controller: starting controller
Mar 20 02:12:08 localhost.localdomain kubelet[28754]: I0320 02:12:08.613549   28754 controller.go:118] kubelet config controller: validating combination of defaults and flags
Mar 20 02:12:08 localhost.localdomain kubelet[28754]: error: unable to load client CA file /etc/kubernetes/pki/ca.crt: open /etc/kubernetes/pki/ca.crt: no such file or directory
Mar 20 02:12:08 localhost.localdomain systemd[1]: kubelet.service: main process exited, code=exited, status=1/FAILURE
Mar 20 02:12:08 localhost.localdomain systemd[1]: Unit kubelet.service entered failed state.
Mar 20 02:12:08 localhost.localdomain systemd[1]: kubelet.service failed.
Mar 20 02:12:10 localhost.localdomain sudo[28759]:  vagrant : TTY=pts/0 ; PWD=/home/vagrant ; USER=root ; COMMAND=/bin/journalctl -xe
```

**Comment by dougbtv** [comment]
@verizonold -- thanks for reporting the issue, and giving it a whirl according to that blog article.

One thing for sure is that blog article is a bit out of date (I have a new one I'm working on that will be published when we finish a refactor of this repository that should be incoming in the next week or so), I'll try to remember to ping you when that's published.

Secondarily -- that blog article recommends pinning the version by doing a `git clone --branch ...` -- there's a possibility that there's a relation to this bug #198 -- which, in summary, states that there's been an upgrade of the Docker version in the CentOS repositories. That issue shows a work-around with upgrading the Kernel and then rebooting your VMs, which might work.

Another possibility is to try to work from the master branch here, and it should be generally similar to the blog article, there have been some changes since then (some major) but, the idea is the same -- spin up virtual machines, and then run the kubernetes install playbook. One thing you'll also have to do now is check out the `README.md` for the instructions to install the external galaxy roles, too, which wasn't implemented yet on the writing of that blog article. The changes in master might have the fix that bug, now, too.

Thanks again @verizonold !

**Comment by leifmadsen** [comment]
Closed due to likely issue being resolved with #198, #200, #203, and #204

**Comment by verizonold** [comment]
@dougbtv  Thanks much!
so, I removed the docker that came with CentOs and installed a new one. Ansible install went through, however I do not see the minions registering with the master. I filed another issue for this.
